### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by Artyomkin6**
PLP & Demodata & PDP

### CODE REVIEW

File: src/main/java/com/es/phoneshop/model/product/PriceHistory.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public BigDecimal getPrice() {
+        return price;
+    }
+
+    public void setPrice(BigDecimal price) {
+        this.price = price;
+    }
+
+    public String getStringDate() {
+        SimpleDateFormat formatter = new SimpleDateFormat("d MMM yyyy");
```
Comment (yan-piatkevich) [inline src/main/java/com/es/phoneshop/model/product/PriceHistory.java]: Declare as constant

File: src/test/java/com/es/phoneshop/model/product/ProductSearchPredicateTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        assertFalse(result);
+    }
+
+    @Test
+    public void testEmptyQueryFoundAllProducts() {
+        String query = "";
+        predicate = new ProductSearchPredicate(query);
+
+        defaultProduct.setDescription("Hello");
```
Comment (yan-piatkevich) [inline src/test/java/com/es/phoneshop/model/product/ProductSearchPredicateTest.java]: Test context preparation should be placed into "when" part of the test or into setUp() method

File: src/test/java/com/es/phoneshop/model/product/ProductSortComparatorTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    @Test
+    public void testDescendingByDescription() {
+        comparator = new ProductSortComparator(
+                SortField.DESCRIPTION, SortOrder.DESCENDING);
+        defaultProduct1.setDescription("Samsung Galaxy S");
+        defaultProduct2.setDescription("Apple iphone");
+
+        int compareResult1 = comparator.compare(defaultProduct1, defaultProduct2);
```
Comment (yan-piatkevich) [inline src/test/java/com/es/phoneshop/model/product/ProductSortComparatorTest.java]: You should use separate @Test for each tested method invocation

File: src/test/java/com/es/phoneshop/web/ProductDetailsPageServletTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.mockito.ArgumentMatchers.*;
```
Comment (yan-piatkevich) [inline src/test/java/com/es/phoneshop/web/ProductDetailsPageServletTest.java]: Expand

File: src/test/java/com/es/phoneshop/web/ProductNotFoundHandlerServletTest.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.mockito.ArgumentMatchers.*;
```
Comment (yan-piatkevich) [inline src/test/java/com/es/phoneshop/web/ProductNotFoundHandlerServletTest.java]: expand

File: src/main/java/com/es/phoneshop/model/product/PriceHistory.java (hunk 1 above)
Comment (Artyomkin6) [reply src/main/java/com/es/phoneshop/model/product/PriceHistory.java]: You mean I should declare as constant here, or make formatter a constant class member?

**Review by yan-piatkevich — APPROVED** [review]

**Pull request merged**